python/cudf/cudf/tests/test_copying.py

```diff
@@ -52,3 +53,309 @@ def test_null_copy():
     col = Series(np.arange(2049))
     col[:] = None
     assert len(col) == 2049
+
+
+def test_series_setitem_cow_on():
+    cudf.set_option("copy_on_write", True)
```

Comment:
TODO: As a new issue, can `cudf.set_option` have context manager behaviour?

Reply:
Yup, done. Here is the issue: https://github.com/rapidsai/cudf/issues/12736